docs/tasks/development-testing.md

```diff
@@ -0,0 +1,42 @@
+# Development Testing
+
+In order to help development for the proxy, there are a few tools in place for
+quick testing.
+
+# Creating a Cluster
+
+Use `make dev_cluster_create` to spin up a kind cluster locally. This will also
+build the proxy and other tooling from source, build their images, and load them
+onto each node.
+
+# Deploying the Proxy
+
+This will build the proxy and other tooling from source,build the images, and
+load them onto each node. This will then deploy the proxy alongside a fake OIDC
+issuer so that the proxy is fully functional. The proxy will then be reachable
+from a node port service in the cluster.
+
+
+```bash
+make dev_cluster_deploy
+```
+
+This command will output a signed OIDC token that is valid for the proxy. You
+can then make calls to the proxy, like the following:
+
+```bash
+curl -k https://172.17.0.2:30226 -H 'Authorization: bearer eyJhbGciOiJSUzI1NiIsImtpZCI6IiJ9.ewoJImlzcyI6Imh0dHBzOi8vb2lkYy1pc3N1ZXItZTJlLmt1YmUtb2lkYy1wcm94eS1lMmUtNmhiNGcuc3ZjLmNsdXN0ZXIubG9jYWw6NjQ0MyIsCgkiYXVkIjpbImt1YmUtb2lkYy1wcm94eS1lMmUtY2xpZW50LWlkIiwiYXVkLTIiXSwKCSJlbWFpbCI6InVzZXJAZXhhbXBsZS5jb20iLAoJImdyb3VwcyI6WyJncm91cC0xIiwiZ3JvdXAtMiJdLAoJImV4cCI6MTU4MjU1NTYzMQoJfQ.qWCM5zUHGslmwbgyZnMjhVeCLJd3R3c7xjtatjT_pv1VY-PpJ8IGBsbcCpur1fAm2CAbr0juM3yzwV1S3TUjhNhE8Wo6rxjA2Flnmwj7Nn2Got6T_cMFHQ_3A6YC72qkMwH-7SvXFB-C5Bk96vi9-clrxJ_b1XjfMPViZEVCJphh9HVzrZ5DPOAR0PDl-qnVys_CRkF0NEwEvAZL5SFumBqjtLBI9XUlWbB6VTljPOExL1zkv8NevZF8DxVsYFaW9HOYH8vNgC07kj_oUVkmAjP-2tVngcBKka0IBmuz2r-RfWNy9VJ-yb19AbtJNw6fjASy7O6VifuH4ZpjP5JSIg'
```

Comment:
So a few nits on this command..

First of all, on OSX the docker network is not routable from the host, due to Docker ultimately running in a (thin) virtual machine. Instead, `localhost` should be used, and `extraPorts` configured when starting the kind cluster to point to a port that is then exposed as a *node port* in the cluster. This will mean your command requires you to `curl` localhost above, and probably with a different port number too.

I did *try* curling the port that is exposed currently by the control plane node:
```
544d3ae0825b        kindest/node:v1.17.0   "/usr/local/bin/entr…"   4 minutes ago       Up 3 minutes        127.0.0.1:32781->6443/tcp   kube-oidc-proxy-e2e-control-plane
```

This fails with:
```
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {

  },
  "status": "Failure",
  "message": "Unauthorized",
  "reason": "Unauthorized",
  "code": 401
}
```

I *think* this is because I'm querying the kube-apiserver directly instead of talking to the proxy, because you haven't set up these port forwards 🙃

Not sure what you'd like to do here - if you want to fix up this bit, let me know and I can test again once you're done (or I can leave you a tmate session running or something so you can test yourself). Otherwise, this PR seems pretty sane & working to me so I'd be happy to just lgtm/approve and open an issue to track this bit (although it seems like you are quite close to having OSX work anyway, so 🤷‍♂)

lgtm'ing, approving, and holding so you can decide yourself :)

/lgtm
/approve
/hold

Reply:
Thanks @munnerz  :) 

I think I will follow up with a PR to fix up the port-forward issue. It is obviously a bug and needs to be dealt with but since lots of other stuff is waiting on this I think it makes sense to get this all through and follow up after. I'll open up an issue.